### PR TITLE
Add risk coordinator to vault

### PR DESCRIPTION
**Pull request by prateekdefi**
Issue: https://linear.app/perennial/issue/PE-1959/[feature]-add-risk-coordinator-to-vault

### CODE REVIEW

File: packages/vault/contracts/Vault.sol [hunk 1]
```diff
@@ -54,6 +54,9 @@ contract Vault is IVault, Instance {
     /// @dev DEPRECATED SLOT -- previously the mappings
     bytes32 private __unused0__;
 
+    /// @dev Risk coordinator of the vault
+    address public coordinator;
```
Comment (EdNoepel) [inline packages/vault/contracts/Vault.sol:58]: Would it be okay to replace the `__unused0__` storage with this?

**Review by EdNoepel — COMMENTED** [review]
One question regarding storage layout, otherwise looks good.

File: packages/vault/contracts/Vault.sol (hunk 1 above)
Comment (prateekdefi) [reply packages/vault/contracts/Vault.sol:58]: okay. i will keep the `__unused0__` storage slot

**Review by EdNoepel — APPROVED** [review]
LGTM

File: packages/vault/contracts/Vault.sol [hunk 2]
```diff
@@ -516,4 +526,10 @@ contract Vault is IVault, Instance {
             );
         }
     }
+
+    /// @notice Only the coordinator can call
+    modifier onlyCoordinator {
+        if (msg.sender != coordinator) revert VaultNotCoordinatorError();
```
Comment (kbrizzle) [inline packages/vault/contracts/Vault.sol:532]: in `Market` we allow the owner to call `onlyCoordinator` functions as well, probably should keep that consistent here.

File: packages/vault/contracts/Vault.sol [hunk 3]
```diff
@@ -216,7 +219,7 @@ contract Vault is IVault, Instance {
 
     /// @notice Settles, then updates the vault parameter set
     /// @param newParameter The new vault parameter set
-    function updateParameter(VaultParameter memory newParameter) external onlyOwner {
+    function updateParameter(VaultParameter memory newParameter) external onlyCoordinator {
```
Comment (kbrizzle) [inline packages/vault/contracts/Vault.sol:222]: Let's keep this one as `onlyOwner` (think of it like the `MarketParameter` of the Vault).

File: packages/vault/contracts/interfaces/IVault.sol [hunk 4]
```diff
@@ -65,6 +66,8 @@ interface IVault is IInstance {
     // sig: 0x50ad85d6
     error VaultCurrentOutOfSyncError();
 
+    error VaultNotCoordinatorError();
```
Comment (kbrizzle) [inline packages/vault/contracts/interfaces/IVault.sol:69]: nit: need to add the signature

**Review by kbrizzle — APPROVED** [review]

**Review by EdNoepel — APPROVED** [review]
LGTM

**Pull request merged**